File: src/main/scala/org/rutiger/theatre/actors/exercise/one/Legolas.scala
package org.rutiger.theatre.actors.exercise.one

import akka.actor.Actor

/*
This class represents a guy that can shots 5 arrows at once. Epicness level over 9k
 */
class Legolas extends Actor{

  //TODO Receive Attack message and send a randome number of arrows, tops to 5.
  override def receive: Receive = ???
}

object Legolas {

  final case class Attack()
  final case class Shot(arrows: Int)

}

File: src/main/java/org/rutiger/theatre/actors/exercise/one/Legolas.java
package org.rutiger.theatre.actors.exercise.one;

import akka.actor.AbstractActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import org.rutiger.theatre.Messages;
import org.rutiger.theatre.actors.exercise.Companion;

/*
This class represents a guy that can shots 5 arrows at once. Epicness level over 9k
 */
public class Legolas extends AbstractActor implements Companion {
    private LoggingAdapter log = Logging.getLogger(getContext().system(), this);

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(Messages.Attack.class, s -> {
                    int arrowsShot = attackWith(5);
                    log.info("shot {} arrows", arrowsShot);
                    getSender().tell(new Messages.Shot(arrowsShot), getSelf());
                })
                .build();
    }
}